docs/reference/api/platform/FileHandle.md

```diff
@@ -0,0 +1,274 @@
+## FileHandle
+
+For general information on [files](file.html) and [filing systems](filesystem.html), see the respective documentation. This chapter covers the abstract API, with an emphasis on devices, as they're less straightforward.
+
+`FileHandle` is an abstract class representing a device that supports file-like operations such as `read` and `write`. This may be an actual `File` on a storage device provided by a `FileSystem`, or a device such as `UARTSerial`.
+
+The `FileHandle` abstraction represents an already-opened file or device, so it has no `open` method of its own - the opening may take the form of a call to another class that returns a `FileHandle`, such as `FileSystem::open`, or it may be implicit in the construction of an object such as `UARTSerial`. 
+
+The `FileHandle` abstraction permits stream-handling code to be device-independent. For example the console input and output streams used for C's `stdin` and `stdout` can be retargeted to something other than the default serial port via the `FileHandle` API, and `ATCmdParser` and the PPP connection to lwIP work on abstract `FileHandle` pointers.
+
+Exactly which operations a `FileHandle` supports will depend on the underlying device, and will in turn restrict what applications it is suitable for. For example, a database application might require random-access and `seek`, but this may not be available on a limited filesystem, and certainly not on a stream device. Only a `File` on a full `FileSystem` such as `FATFileSystem` would generally implement the entire API. Specialised devices may have particular limitations or behavior, which they should document, and which would limit their general utility.
+
+### Relationship of FileHandle to other APIs
+
+Mbed `FileHandle`s can be used directly, but they are often manipulated via POSIX or C/C++ APIs. The layering is that stdio calls taking `FILE *stream` call the POSIX APIs taking `int fd`, which call methods on `FileHandle` objects.
+
```

Comment:
It may help to add a picture here. Maybe something like this?

``` cpp
C stdio API:                  FILE *fopen(const char *path, const char *mode);
                                      |
                                      v
POSIX API:                    int open(const char *path, int flags);
                                      |
                                      v
Mbed OS C++ FileHandle API:   int FileHandle::open(const char *path, int flags);
                                      |
                                      v
Mbed OS C++ File:             int File::open(FileSystem *fs, const char *path, int flags);
                                      |
                                      v
Mbed OS C++ FileSystem:       int FileSystem::file_open(mbed::fs_file_t *f, const char *path, int flags);
                                      |
                                      v
Mbed OS C++ FATFileSystem:    int FATFileSystem::file_open(mbed::fs_file_t *f, const char *path, int flags);
                                      |
                                      v
ChanFS FAT File System:       FRESULT f_open (FIL* fp, const TCHAR* path, BYTE mode);
```

Ok I may have gotten a bit carried away. Feel free to cut that off where it's relevant.

Reply:
Probably helps. Think we fall out of scope after `File` - a concrete implementation of `FileHandle`.

And `open` is a special case. Maybe stick to a core call.